artix7/primitives/ff/ff.pb_type.xml

```diff
@@ -0,0 +1,239 @@
+          <output name="Q" num_pins="1"/>
+          <T_setup    value="10e-12" port="BEL_FF-FDPE.D" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDPE.CE" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDPE.PRE" clock="C" />
+          <T_clock_to_Q max="10e-12" port="BEL_FF-FDPE.Q" clock="C" />
+        </pb_type>
+
+        <interconnect>
+          <direct name="D" input="BEL_FF-FDPE_or_FDCE.D" output="BEL_FF-FDPE.D" />
+          <direct name="CE" input="BEL_FF-FDPE_or_FDCE.CE" output="BEL_FF-FDPE.CE" />
+          <direct name="C" input="BEL_FF-FDPE_or_FDCE.C" output="BEL_FF-FDPE.C" />
+          <direct name="PRE" input="BEL_FF-FDPE_or_FDCE.SR" output="BEL_FF-FDPE.PRE" />
+          <direct name="Q" input="BEL_FF-FDPE.Q" output="BEL_FF-FDPE_or_FDCE.Q" />
+        </interconnect>
+      </mode>
+      <mode name="FDCE">
+        <pb_type name="BEL_FF-FDCE" num_pb="1" blif_model=".subckt FDCE">
+          <input  name="D" num_pins="1"/>
+          <input  name="CE" num_pins="1"/>
+          <clock  name="C" num_pins="1"/>
+          <input  name="CLR" num_pins="1"/>
+          <output name="Q" num_pins="1"/>
+          <T_setup    value="10e-12" port="BEL_FF-FDCE.D" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDCE.CE" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDCE.CLR" clock="C" />
+          <T_clock_to_Q max="10e-12" port="BEL_FF-FDCE.Q" clock="C" />
+        </pb_type>
+
+        <interconnect>
+          <direct name="D" input="BEL_FF-FDPE_or_FDCE.D" output="BEL_FF-FDCE.D" />
+          <direct name="CE" input="BEL_FF-FDPE_or_FDCE.CE" output="BEL_FF-FDCE.CE" />
+          <direct name="C" input="BEL_FF-FDPE_or_FDCE.C" output="BEL_FF-FDCE.C" />
+          <direct name="CLR" input="BEL_FF-FDPE_or_FDCE.SR" output="BEL_FF-FDCE.CLR" />
+          <direct name="Q" input="BEL_FF-FDCE.Q" output="BEL_FF-FDPE_or_FDCE.Q" />
+        </interconnect>
+
+        <metadata>
+          <meta name="fasm_features">ZRST</meta>
+        </metadata>
+      </mode>
+      <metadata>
+        <meta name="fasm_prefix">AFF BFF CFF DFF A5FF B5FF C5FF D5FF</meta>
+      </metadata>
+    </pb_type>
+
+    <interconnect>
+      <complete name="CE" input="BLK_BB-SLICE_FF.CE" output="BEL_FF-FDPE_or_FDCE.CE" />
+      <complete name="C" input="BLK_BB-SLICE_FF.CK" output="BEL_FF-FDPE_or_FDCE.C" />
+      <complete name="SR" input="BLK_BB-SLICE_FF.SR" output="BEL_FF-FDPE_or_FDCE.SR" />
+
+      <direct name="D" input="BLK_BB-SLICE_FF.D[3:0]" output="BEL_FF-FDPE_or_FDCE[3:0].D" />
+      <direct name="Q" input="BEL_FF-FDPE_or_FDCE[3:0].Q" output="BLK_BB-SLICE_FF.Q[3:0]" />
+
+      <direct name="D5" input="BLK_BB-SLICE_FF.D5[3:0]" output="BEL_FF-FDPE_or_FDCE[7:4].D" />
+      <direct name="Q5" input="BEL_FF-FDPE_or_FDCE[7:4].Q" output="BLK_BB-SLICE_FF.Q5[3:0]" />
+    </interconnect>
+  </mode>
+  <mode name="LDPE_or_LDCE">
+    <pb_type name="BEL_FF-LDPE_or_LDCE" num_pb="8">
```

Comment:
I was pretty sure that only *some* of the FFs supported working in latch mode?

Reply:
![image](https://user-images.githubusercontent.com/21212/47124401-836da980-d232-11e8-85fa-758bd6b0745c.png)

Reply:
Yep, that is correct.  Sad :(